internal/storage/driver/adapters.go

```diff
@@ -20,12 +20,12 @@ func (d *DefaultStorageDriverAdapter) OpenLedger(ctx context.Context, name strin
 		return nil, nil, err
 	}
 
-	isUpToDate, err := store.GetBucket().IsUpToDate(ctx, d.db)
+	lastVersion, err := store.GetBucket().GetLastVersion(ctx, d.db)
 	if err != nil {
 		return nil, nil, fmt.Errorf("checking if bucket is up to date: %w", err)
 	}
 
-	return ledgerstore.NewDefaultStoreAdapter(isUpToDate, store), l, nil
+	return ledgerstore.NewDefaultStoreAdapter(lastVersion >= 26, store), l, nil
```

Comment:
Magical number, corresponding to the migration 25.
The migration 25 was introduced in v2.2.0 of the ledger.
So checking for this version is enough to declare the database is up to date to use the new store.